js/app/modules/contentGroup/contentGroup.js

```diff
@@ -267,6 +271,10 @@ const ContentGroup = new Module.Class({
 
     },
 
+    _on_reading_history_changed: function () {
+        this._arrangement.mark_read(ReadingHistoryModel.get_default().get_read_articles());
```

Comment:
At first glance I would rather see the models being marked read, and then that change would propagate to the cards. 

Come to think of it, I would have rather seen this implemented as per-user EKN metadata which has been on our list for quite a long time, but I don't think we'll get around to it in 3.1. In any case we should do something here that's easy to refactor in that direction.